src/components/TodoComponents/TodoList.js

```diff
@@ -1,2 +1,302 @@
+            a.setAttribute("id", this.id + "autocomplete-list");
+            a.setAttribute("class", "autocomplete-items");
+
+            this.parentNode.appendChild(a);
+
+            for (i = 0; i < items.length; i++) {
+
+                if (items[i].substr(0, val.length).toUpperCase() == val.toUpperCase()) {
+
+                    b = document.createElement("DIV");
+
+                    b.innerHTML = "<strong>" + items[i].substr(0, val.length) + "</strong>";
+                    b.innerHTML += items[i].substr(val.length);
+
+                    b.innerHTML += "<input type='hidden' value='" + items[i] + "'>";
+
+                    b.addEventListener("click", function (e) {
+
+                        inp.value = this.getElementsByTagName("input")[0].value;
+
+                        closeAllLists();
+                    });
+                    a.appendChild(b);
+                }
+            }
+        });
+
+        inp.addEventListener("keydown", function (e) {
+            var x = document.getElementById(this.id + "autocomplete-list");
+            if (x) x = x.getElementsByTagName("div");
+            if (e.keyCode == 40) {
+
+                currentFocus++;
+
+                addActive(x);
+            } else if (e.keyCode == 38) {
+                currentFocus--;
+
+                addActive(x);
+            } else if (e.keyCode == 13) {
+
+                e.preventDefault();
+                if (currentFocus > -1) {
+
+                    if (x) x[currentFocus].click();
+                }
+            }
+        });
+        function addActive(x) {
+
+            if (!x) return false;
+
+            removeActive(x);
+            if (currentFocus >= x.length) currentFocus = 0;
+            if (currentFocus < 0) currentFocus = (x.length - 1);
+
+            x[currentFocus].classList.add("autocomplete-active");
+        }
+        function removeActive(x) {
+
+            for (var i = 0; i < x.length; i++) {
+                x[i].classList.remove("autocomplete-active");
+            }
+        }
+        function closeAllLists(elmnt) {
+
+            var x = document.getElementsByClassName("autocomplete-items");
+            for (var i = 0; i < x.length; i++) {
+                if (elmnt != x[i] && elmnt != inp) {
+                    x[i].parentNode.removeChild(x[i]);
+                }
+            }
+        }
+
+        document.addEventListener("click", function (e) {
+            closeAllLists(e.target);
+        });
+    }
+
+
+render() {
+
+
+    return (
+        <div className="todoListMain">
+            <div className='header'>
+                <form autocomplete="off" action="/action_page.php" onSubmit={this.addItem} >
+                    <input ref={(a) => this._inputElement = a}
+                        placeholder="Enter ToDo item">
+                    </input>
+                    <button type="submit" >Add ToDo</button>
+                    <button onClick={this.deleteItem}>Clear Completed</button>
```

Comment:
See how you have attached a method to an event using your JSX here? That's what you should have done with all of your pomodoro stuff. That is the react way of handling it.